holoviews/operation/datashader.py

```diff
@@ -0,0 +1,208 @@
+    transfer functions is not yet supported.
+    """
+    x0, x1 = canvas.x_range
+    y0, y1 = canvas.y_range
+    kdims = [dataset.get_dimension(d) for d in (glyph.x, glyph.y)]
+
+    column = summary.column
+    if column and isinstance(summary, ds.count_cat):
+        name = '%s Count' % summary.column
+    else:
+        name = column
+    vdims = [dataset.get_dimension(column)(name) if column
+             else Dimension('Count')]
+
+    aggregate = pandas_pipeline(dataset.dframe(), schema, canvas,
+                                glyph, summary)
+    aggregate = aggregate.rename({'x_axis': kdims[0].name,
+                                  'y_axis': kdims[1].name})
+
+    params = dict(get_param_values(dataset), kdims=kdims,
+                  datatype=['xarray'], vdims=vdims)
+
+    if aggregate.ndim == 2:
+        return GridImage(aggregate, **params)
+    else:
+        return NdOverlay({c: GridImage(aggregate.sel(**{column: c}),
+                                       **params)
+                          for c in aggregate.coords[column].data},
+                         kdims=[dataset.get_dimension(column)])
+
+
+class Aggregate(ElementOperation):
+
+    aggregator = param.ClassSelector(class_=ds.reductions.Reduction,
+                                     default=ds.count())
+
+    height = param.Integer(default=800)
+
+    width = param.Integer(default=600)
+
+    x_range  = param.NumericTuple(default=None, length=2,
+                                  allow_None=True)
+
+    y_range  = param.NumericTuple(default=None, length=2,
+                                  allow_None=True)
+
+    x_sampling = param.Number(default=None, doc="""
+        Specifies the smallest allowed sampling interval along the y-axis.""")
+
+    y_sampling = param.Number(default=None, doc="""
+        Specifies the smallest allowed sampling interval along the y-axis.""")
+
+    @classmethod
+    def get_agg_data(cls, obj, category=None):
+        paths = []
+        kdims = obj.kdims
+        vdims = obj.vdims
+        x, y = obj.dimensions(label=True)[:2]
+        if isinstance(obj, Path):
+            line = True
+            for p in obj.data:
+                df = pd.DataFrame(p, columns=obj.dimensions('key', True))
+                if isinstance(obj, Contours) and obj.vdims and obj.level:
+                    df[obj.vdims[0].name] = p.level
+                paths.append(df)
+        elif isinstance(obj, CompositeOverlay):
+            for key, el in obj.data.items():
+                x, y, element, line = cls.get_agg_data(el)
+                df = element.dframe()
+                if isinstance(obj, NdOverlay):
+                    df = df.assign(**dict(zip(obj.dimensions('key', True), key)))
+                paths.append(df)
+            kdims += element.kdims
+            vdims = element.vdims
+        elif isinstance(obj, Element):
+            line = isinstance(obj, Curve)
+            paths.append(obj.dframe())
+        if line:
```

Comment:
It's never nice to see a bunch of if-else blocks to support different types, but I don't see a cleaner way to propose.